src/libraries/Microsoft.CSharp/src/ILLinkTrim.xml

```diff
@@ -4,5 +4,7 @@
     <type fullname="Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView"/>
     <!-- Required by the Expression Evaluator -->
     <type fullname="Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView/DynamicDebugViewEmptyException"/>
+    <!-- Required for COM event dispatch -->
+    <type fullname="System.Runtime.InteropServices.ComEventsSink"/>
```

Comment:
This prevents ComEventsSink in Microsoft.CSharp.dll from being trimmed out.  Is it being accessed by reflection somewhere and that's why this is needed?

Reply:
Parts of it will be accessed by whatever COM server is firing the events for which a handler has been added. Specifically, it is the `ComEventsSink`'s implementation of `IDispatch` that is not explicitly used (directly or via reflection), but is necessary for handling COM events.

Is there a better way to prevent that from being trimmed out? I know there is `PreserveDependency`, but since we are not actually the ones that call into it, we don't actually have another method that is using it.

Reply:
Thanks for the explanation.  I think what you have is the best we can currently do.  That said, are you actually seeing the interface implementation get removed?  I didn't think the linker would do that if the interface is used anywhere, but my knowledge may be out of date.

Reply:
Yeah, I hit this because the implementation of `IDispatch.Invoke` was getting removed.

Reply:
https://github.com/mono/linker/issues/936 ?